MONO: Structure Dissect datatype correction for Enum as Array element + Funcs for C# class "Properties" manipulation

PeaceBeUponYou opened the pull request.

PeaceBeUponYou committed:
Structure Dissect correction for Array when elements are Enum values

PeaceBeUponYou committed:
Merge branch 'master' of https://github.com/PeaceBeUponYou/cheat-engine

PeaceBeUponYou committed:
Added Methods to manipulate "Properties" of a class

PeaceBeUponYou committed:
Fix for multi dimensional arrays in Structure Dissect

PeaceBeUponYou committed:
Added a method to get MonoClass* from ANY MonoType*

Turns out mono_type_get_class was designed only for MONO_TYPE_CLASS and MONO_TYPE_VALUETYPE

PeaceBeUponYou committed:
Merge branch 'cheat-engine:master' into master

PeaceBeUponYou commented:
This new error has nothing to do with the changes I made. You will need to manually debug the function "mono_class_getVTable" to figure out where it is causing the problem.

PeaceBeUponYou committed:
dotnet patch fixes when injecting a C# assembly created by CSCompile

The person who did last editing did not really test the changes at all.

PeaceBeUponYou committed:
New methods & Merging changes from CE git changelog

Merged Methods:
*mono_class_getNestedTypes
*mono_method_getFlags

New Methods:
*mono_method_is_generic
*mono_method_is_inflated
*mono_method_is_instance
*mono_gchandle_new
*mono_gchandle_new_weakref
*mono_gchandle_get_target
*mono_gchandle_free
*mono_gchandle_free_all
*mono_class_enumEvents
*mono_event_get_name
*mono_event_get_parent
*mono_event_get_add_method
*mono_event_get_remove_method
*mono_event_get_raise_method
*mono_event_get_object
*mono_event_get_flags
*mono_class_is_blittable
*mono_class_instance_size
*mono_type_get_attrs

cheat-engine commented:
> specifically when trying to access lists/dicts of an in-game inventory of resources and their amount

finally got to go through the monoscript file.  Just change https://github.com/cheat-engine/cheat-engine/blob/master/Cheat%20Engine/bin/autorun/monoscript.lua#L1724 to readDword()

PeaceBeUponYou committed:
Fixes for Il2cpp in mono_string_new and monosymbolist ittr

PeaceBeUponYou commented:
It returns a pointer, so should it not be readQword()? Also it is readQword in both cpp and lua files

PeaceBeUponYou committed:
fix in "mono_array_element_size"

PeaceBeUponYou commented:
Done

PeaceBeUponYou committed:
Added methods for static field data

Added:
 "mono_class_get_static_field_data" --basically returns the base address of static fields in both mono and il2cpp
Fix:
 in "mono_class_getStaticFieldAddress", now returns the base address for il2cpp too

PeaceBeUponYou committed:
A Method to register an il2cpp function to full length

PeaceBeUponYou committed:
Added support for reference type returntypes in mono_invoke_method (value will return as pointer)
Updated version to the date

PeaceBeUponYou committed:
Added support for reference type returntypes in mono_invoke_method (value will return as pointer)
Updated version to the date

PeaceBeUponYou committed:
Merge branch 'master' of https://github.com/PeaceBeUponYou/cheat-engine
